pkg/artifactory/repositories.go

```diff
@@ -1132,11 +1241,231 @@ func mkResourceSchema(skeema map[string]*schema.Schema, packer PackFunc, unpack
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 
-		Schema: skeema,
+		Schema:        skeema,
 		CustomizeDiff: projectEnvironmentsDiff,
 	}
 }
 
+//Returns random string from a map[string]string
+func selectRandomFromMapOfStrings(m map[string]string) string {
+	mapLength := len(m)
+	allValues := make([]string, 0, mapLength)
+	for _, value := range m {
+		allValues = append(allValues, value)
+	}
+	s1 := rand.NewSource(time.Now().UnixNano())
+	r1 := rand.New(s1)
+	return allValues[r1.Intn(mapLength)]
+}
+
+func isSelectRandom(opts ...bool) bool {
+	selectRandomFlag := false
+	for i, val := range opts {
+		switch i {
+		case 0:
+			selectRandomFlag = val
+		default:
+			fmt.Printf("Option index is not defined. Index: %v, value: %v\n", i, val)
+		}
+	}
+	return selectRandomFlag
+}
+
+//Return the default repo layout of Local Repository type
+func getDefaultLocalRepoLayoutRef(repoType string, opts ...bool) string {
+
+	selectRandom := isSelectRandom(opts...)
+	const defaultRepoLayout = "simple-default"
+	repoLayout := defaultRepoLayout
+
+	localDefaultRepoLayoutMap := map[string]string{
+		"alpine":    "simple-default",
+		"bower":     "bower-default",
+		"cran":      "simple-default",
+		"cargo":     "simple-default",
+		"chef":      "simple-default",
+		"cocoapods": "simple-default",
+		"composer":  "composer-default",
+		"conan":     "conan-default",
+		"conda":     "simple-default",
+		"debian":    "simple-default",
+		"docker":    "simple-default",
+		"gems":      "simple-default",
+		"generic":   "simple-default",
+		"gitlfs":    "simple-default",
+		"go":        "go-default",
+		"gradle":    "maven-2-default",
+		"helm":      "simple-default",
+		"ivy":       "ivy-default",
+		"maven":     "maven-2-default",
+		"npm":       "npm-default",
+		"nuget":     "nuget-default",
+		"opkg":      "simple-default",
+		"pub":       "simple-default",
+		"puppet":    "puppet-default",
+		"pypi":      "simple-default",
+		"sbt":       "sbt-default",
+		"vagrant":   "simple-default",
+		"rpm":       "simple-default",
+	}
+
+	if selectRandom {
+		repoLayout = selectRandomFromMapOfStrings(localDefaultRepoLayoutMap)
+	} else {
+		if value, ok := localDefaultRepoLayoutMap[repoType]; ok {
+			repoLayout = value
+		}
+	}
+	return repoLayout
+}
+
+//Return the default repo layout of Remote Repository type
+func getDefaultRemoteRepoLayoutRef(repoType string, opts ...bool) string {
```

Comment:
Please don't do this. In fact, this whole function is unnecessary - all you need is this map - you don't need any function. 

And, I wouldn't even have a default - if your lookup fails, then panic because it's a type we don't support

Reply:
Improved. new function is `getDefaultRepoLayoutRef`